File: gdx2d-demoDesktop/src/ch/hevs/gdx2d/demos/simple/DemoSimpleShapes.java
package ch.hevs.gdx2d.demos.simple;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.math.Vector2;
import hevs.gdx2d.components.graphics.Polygon;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * A very simple demonstration on how to
 * display something with the library
 *
 * @author Pierre-André Mudry (mui)
 * @version 1.0
 */
public class DemoSimpleShapes extends PortableApplication {

	public static void main(String args[]) {
		new DemoSimpleShapes();
	}

	@Override
	public void onInit() {
		this.setTitle("Simple shapes, mui 2013");
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {
		g.clear();

		// Draws a yellow circle
		g.setColor(Color.YELLOW);
		g.drawCircle(250, 400, 20);

		// Draws a green rectangle
		g.setColor(Color.GREEN);
		g.drawRectangle(20, 250, 40, 40, 0);

		g.drawFilledCircle(50, 50, 20, Color.PINK);
		g.drawFilledRectangle(80, 30, 20, 20, 0, new Color(0.5f, 0.5f, 0.5f, 0.5f));

		// Draws a blue polygon
		Vector2 points[] = {
				new Vector2(200, 200),
				new Vector2(250, 250),
				new Vector2(300, 200)
		};

		Polygon p = new Polygon(points);
		g.drawFilledPolygon(p, Color.BLUE);
	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/simple/DemoSimpleShapes.scala
package ch.hevs.gdx2d.demos.simple

import ch.hevs.gdx2d.components.graphics.Polygon
import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.math.Vector2

/**
 * A very simple demonstration on how to display something with the library.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoSimpleShapes extends DesktopApplication {

  override def onInit(): Unit = {
    setTitle("Simple shapes, mui 2013")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    g.clear()

    // Draws a yellow circle
    g.setColor(Color.YELLOW)
    g.drawCircle(250f, 400f, 20f)

    // Draws a green rectangle
    g.setColor(Color.GREEN)
    g.drawRectangle(20f, 250f, 40f, 40f, 0f)

    g.drawFilledCircle(50f, 50f, 20f, Color.PINK)
    g.drawFilledRectangle(80f, 30f, 20f, 20f, 0f, new Color(0.5f, 0.5f, 0.5f, 0.5f))

    // Draws a blue polygon
    val points = Array(new Vector2(200f, 200f), new Vector2(250f, 250f), new Vector2(300f, 200f))
    val p = new Polygon(points)
    g.drawFilledPolygon(p, Color.BLUE)
  }
}

object DemoSimpleShapes {
  def main(args: Array[String]): Unit = {
    new DemoSimpleShapes().launch()
  }
}
